DNN Platform/DotNetNuke.DependencyInjection/Extensions/BuildUpExtensions.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+namespace DotNetNuke.DependencyInjection.Extensions
+{
+    using System;
+    using System.Linq;
+    using System.Reflection;
+
+    /// <summary>
+    /// Adds property injection extension methods.
+    /// </summary>
+    public static class BuildUpExtensions
+    {
+        /// <summary>
+        /// Injects property dependency for public or proetected
+        /// properties that are decorated with <see cref="DependencyAttribute"/>.
+        /// </summary>
+        /// <param name="container">The service provider.</param>
+        /// <param name="myObject">The object to inject properties.</param>
+        public static void BuildUp(this IServiceProvider container, object myObject)
+        {
+            if (container == null || myObject == null)
+            {
+                return;
+            }
+
+            var properties = myObject
+                .GetType()
+                .GetProperties(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic)
```

Comment:
It might be worth profiling whether caching would help here.  If we're doing this many times per request, looking up the properties to fill might become a hot path to avoid.

Reply:
Won't caching the objects cause side-effects if 2 different users make the same request? Wouldn't user 1 get the scoped service from user 2 or vice-versa?

Reply:
I'm thinking about caching the property lookup, not the object itself.

```c#
var properties = GetDependencyProperties(myObject.GetType());

private IEnumerable<PropertyInfo> GetDependencyProperties(Type type)
{
    return CBO.GetCachedObject<IEnumerable<PropertyInfo>(
        new CacheItemArgs($"DependencyProperties_{type.TypeFullName}"),
        (args) => 
            type
                .GetProperties(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic)
                .Where(propertyInfo => propertyInfo.GetCustomAttribute<DependencyAttribute>() != null)
                .ToList());
}
```

Reply:
I have added CBO caching in the latest code push